internal/kv/mem/mem_kv.go

```diff
@@ -217,7 +219,10 @@ func (kv *MemoryKV) MultiRemove(keys []string) error {
 }
 
 // MultiSaveAndRemove saves and removes given key-value pairs in MemoryKV atomicly.
-func (kv *MemoryKV) MultiSaveAndRemove(saves map[string]string, removals []string) error {
+func (kv *MemoryKV) MultiSaveAndRemove(saves map[string]string, removals []string, preds ...predicates.Predicate) error {
+	if len(preds) > 0 {
+		return merr.WrapErrServiceUnavailable("predicates not supported")
```

Comment:
maybe return a NotImplement error is better 🤔

Reply:
Mem KV does not support transaction feature
according to @yah01 this shall use ServiceUnavailable